imap_processing/ena_maps/utils/map_utils.py

```diff
@@ -69,20 +75,25 @@ def bin_single_array_at_indices(
             " projection indices."
         )
 
+    input_valid_mask = np.asarray(input_valid_mask, dtype=bool)
+    mask_idx = input_valid_mask[input_indices]
+
     num_projection_indices = np.prod(projection_grid_shape)
 
+    # Only valid values are summed into bins.
     if value_array.ndim == 1:
+        values = value_array[input_indices]
         binned_values = np.bincount(
-            projection_indices,
-            weights=value_array[input_indices],
+            projection_indices[mask_idx],
+            weights=values[mask_idx],
             minlength=num_projection_indices,
         )
```

Comment:
I find this a bit confusing. I think I have myself convinced it is doing the right thing but was wondering if you find this clearer:
```suggestion
    input_valid_mask = np.asarray(input_valid_mask, dtype=bool)

    num_projection_indices = np.prod(projection_grid_shape)

    # Only valid values are summed into bins.
    if value_array.ndim == 1:
        binned_values = np.bincount(
            projection_indices[input_valid_mask],
            weights=value_array[input_indices[input_valid_mask]],
            minlength=num_projection_indices,
        )
```

Reply:
Yes. That's cleaner.